### PR TITLE
build: add MegaLinter config files

### LINKED ISSUE #1944: Use MegaLinter?
Since this repository contains files in quite a few different languages, setting up MegaLinter could provide many suggestions.
https://megalinter.io/latest/

There is some overlap in role with the pre-commit hook, though.

### PR DESCRIPTION
**Pull request by eitsupi**
Close #1944

- Automatic fixes are disabled to avoid conflicts with existing pre-commits.
- The linters for languages (Rust, JS, and Python) which are already configured (?) are disabled.

### CODE REVIEW

File: .github/workflows/mega-linter.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        if: ${{ success() }} || ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: MegaLinter reports
+          path: |
+            megalinter-reports
+            mega-linter.log
+
+      # Create pull request if applicable (for now works only on PR from same repository, not from forks)
+      - name: Create Pull Request with applied fixes
```
Comment (max-sixty) [inline .github/workflows/mega-linter.yaml:55]: Edit: removed comment as I worked out how it was operating, sorry for the noise

Comment (max-sixty) [reply .github/workflows/mega-linter.yaml:55]: Ah but actually is this consistent with https://github.com/PRQL/prql/pull/1960/files#diff-32bb34bec0def972aa10e47d2903d57917f5fd3834222b4a2bb7c925e6aa0990R15 ?

**Comment by max-sixty** [comment]
I had a look through the checks. Some are quite strict, in a way that is actually quite helpful, but we wouldn't want to completely block code — for example:

```
  Clone found (rust):
   - prql-compiler/tests/integration/main.rs [178:25 - 193:9] (15 lines, 108 tokens)
     prql-compiler/tests/integration/main.rs [103:25 - 118:7]
```

---

In order to fully test it, we might want to open a branch on this repo — it's trying to do things like add comments to the PR, and failing because it's on a fork. We could get around that later by having it run on `pull-request-target`, which gives more permissions but runs based on the code on `main`, not the proposed PR.

@eitsupi you should have permissions to push a branch to this repo.

---

I don't see it making changes to Java code etc? 

For me, there are two helpful things from these:
- Autofixing things that don't change semantics, like formatting
- Giving _information_ about potential bad code, but not stopping merging — semantics are up to the contributor, and we shouldn't slow anything down

One advantage of this was to run formatters on code we don't use much, like Java & C (and maybe something like python, which hardly exists in the repo). But maybe those don't actually have formatters?

**Comment by eitsupi** [comment]
@max-sixty Thanks for taking a look at this.
You are right, I probably should have pushed the branch to this repository instead of my fork...
To see how comments work, please see here as I opened a PR from the same branch on the fork. eitsupi/prql#2

> I don't see it making changes to Java code etc?

So far only the linter corresponding to the fixes included in the PR is working.
We could have configured it to inspect the entire project, but I forgot how to do it so I will check the documentation later to demonstrate that......

> But maybe those don't actually have formatters?

While the ability exists for some linters or formatters to automatically correct, but auto-fixes are completely stopped to avoid conflicts with pre-commits CI.
Users will need to manually fix them.

If we want MegaLinter to automatically fix your problem, we need to stop the pre-commit auto-fix.
(I am fine with either. I have no strong opinion.)

> I had a look through the checks. Some are quite strict, in a way that is actually quite helpful, but we wouldn't want to completely block code — for example:

We need to determine which Linter to disable, or to generate a warning but not treat it as an error.....
The initial setup is quite labor intensive because of the need to look at the settings for a large number of linters.
(This is why I have not been able to implement MegaLinter in most of my projects.)

**Comment by eitsupi** [comment]
@max-sixty Could you please tell me why pre-commit CI is failing in this PR?
Sorry, but I can't make sense of the error message.

**Comment by max-sixty** [comment]
> @max-sixty Could you please tell me why pre-commit CI is failing in this PR? Sorry, but I can't make sense of the error message.

It's not permitted to fix workflow files by GitHub. 

(at the bottom it states "GitHub prevented pre-commit.ci from autofixing this pr due to autofixes to a workflow file")

So we need to run locally and push the result. Lmk if you don't have it installed and I'm happy to do it

**Comment by eitsupi** [comment]
The last commit made all files subject to Lint and found that it takes more than 5000 seconds (!) to check CSS.
Clearly this Linter needs to be disabled.

```log
+----SUMMARY----+--------------------------+---------------+-------+-------+--------+--------------+
| Descriptor    | Linter                   | Mode          | Files | Fixed | Errors | Elapsed time |
+---------------+--------------------------+---------------+-------+-------+--------+--------------+
| ❌ ACTION     | actionlint               | list_of_files |    16 |       |      2 |        0.19s |
| ✅ BASH       | bash-exec                | file          |     2 |       |      0 |        0.01s |
| ❌ BASH       | shellcheck               | list_of_files |     2 |       |      1 |        0.06s |
| ◬ BASH        | shfmt                    | list_of_files |     2 |       |      1 |        0.03s |
| ❌ C          | cpplint                  | list_of_files |     2 |       |      6 |        0.21s |
| ◬ COPYPASTE   | jscpd                    | project       |   n/a |       |     11 |       17.46s |
| ❌ CPP        | cpplint                  | list_of_files |     1 |       |      2 |        0.14s |
| ◬ CSHARP      | csharpier                | list_of_files |     4 |       |      1 |        1.57s |
| ◬ CSHARP      | dotnet-format            | file          |     4 |       |      4 |        2.79s |
| ❌ CSS        | stylelint                | list_of_files |    12 |       |      1 |     5138.48s |
| ❌ DOCKERFILE | hadolint                 | list_of_files |     1 |       |      1 |        0.09s |
| ❌ HTML       | djlint                   | list_of_files |    19 |       |     16 |        1.07s |
| ❌ HTML       | htmlhint                 | list_of_files |    19 |       |     75 |        0.55s |
| ❌ JAVA       | checkstyle               | list_of_files |     4 |       |     76 |        3.09s |
| ❌ JAVA       | pmd                      | list_of_files |     4 |       |      1 |        3.17s |
| ✅ JSON       | eslint-plugin-jsonc      | list_of_files |     9 |       |      0 |        3.07s |
| ❌ JSON       | jsonlint                 | list_of_files |     9 |       |      1 |        0.27s |
| ✅ JSON       | prettier                 | list_of_files |     9 |       |      0 |         1.7s |
| ✅ JSON       | v8r                      | list_of_files |     9 |       |      0 |       10.87s |
| ❌ MAKEFILE   | checkmake                | file          |     1 |       |      1 |         0.0s |
| ◬ MARKDOWN    | markdownlint             | list_of_files |    97 |       |     48 |        1.04s |
| ✅ MARKDOWN   | markdown-link-check      | list_of_files |    97 |       |      0 |       53.58s |
| ◬ MARKDOWN    | markdown-table-formatter | list_of_files |    97 |       |      1 |        0.38s |
| ❌ PHP        | phpcs                    | list_of_files |     3 |       |      1 |        0.15s |
| ✅ PHP        | phplint                  | list_of_files |     3 |       |      0 |        0.09s |
| ❌ PHP        | phpstan                  | list_of_files |     3 |       |      9 |        2.48s |
| ❌ PHP        | psalm                    | list_of_files |     3 |       |     11 |        0.62s |
| ◬ REPOSITORY  | checkov                  | project       |   n/a |       |      6 |       16.52s |
| ◬ REPOSITORY  | devskim                  | project       |   n/a |       |      3 |        1.23s |
| ✅ REPOSITORY | dustilock                | project       |   n/a |       |      0 |        1.28s |
| ✅ REPOSITORY | gitleaks                 | project       |   n/a |       |      0 |        3.75s |
| ✅ REPOSITORY | git_diff                 | project       |   n/a |       |      0 |        0.05s |
| ✅ REPOSITORY | secretlint               | project       |   n/a |       |      0 |         3.3s |
| ✅ REPOSITORY | syft                     | project       |   n/a |       |      0 |        2.17s |
| ◬ REPOSITORY  | trivy                    | project       |   n/a |       |      1 |        9.96s |
| ❌ SPELL      | misspell                 | list_of_files |   201 |       |      1 |        0.35s |
| ✅ SQL        | sql-lint                 | file          |     1 |       |      0 |        0.38s |
| ❌ SQL        | tsqllint                 | list_of_files |     1 |       |      1 |         0.5s |
| ✅ XML        | xmllint                  | list_of_files |     1 |       |      0 |         0.0s |
| ◬ YAML        | prettier                 | list_of_files |    30 |       |      1 |        1.74s |
| ❌ YAML       | v8r                      | list_of_files |    30 |       |      1 |       36.88s |
| ✅ YAML       | yamllint                 | list_of_files |    30 |       |      0 |        0.71s |
+---------------+--------------------------+---------------+-------+-------+--------+--------------+
```

Open another PR (#1974) and continue this.

**Pull request closed without merging**